src/main/java/org/openrewrite/maven/ux/ProgressBarParsingEventListener.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven.ux;
+
+import org.openrewrite.Parser;
+import org.openrewrite.ProgressBar;
+import org.openrewrite.SourceFile;
+import org.openrewrite.tree.ParsingEventListener;
+
+public class ProgressBarParsingEventListener implements ParsingEventListener {
+
+    private final ProgressBar progressBar;
+
+    public ProgressBarParsingEventListener(ProgressBar progressBar) {
+        this.progressBar = progressBar;
+    }
+
+    @Override
+    public void parsed(Parser.Input input, SourceFile sourceFile) {
+        progressBar.intermediateResult("Parsed " + input.getPath());
```

Comment:
Would we want to add any details here based on the produced SourceFile or ParseError?

Reply:
This is incorrect anyway, as it is called _after_ the file is parsed.